tests/unit/nupic/algorithms/knn_classifier_test.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
```

Comment:
Yes the kNN classifier needs unit tests, and we also need to [redo the integration tests](https://github.com/numenta/nupic/issues/2172). The current integration tests do run and test the kNN under different scenarios, not just w/ PCA @david-ragazzi, but they're poorly written/formatted.